kernel/src/engine/arrow_conversion.rs

```diff
@@ -251,6 +254,7 @@ impl TryFrom<&ArrowDataType> for DataType {
                     panic!("DataType::Map should contain a struct field child");
                 }
             }
+            ArrowDataType::Dictionary(_, value_type) => Ok(value_type.as_ref().try_into()?),
```

Comment:
nit: might want a comment to explain that a `Dictionary` type is just the same as its value type wrt. to the schema